utils/superset/db_name_change.py

```diff
@@ -0,0 +1,101 @@
+
+
+def file_handle_error(function):
+    """
+    Decorator to handle the exceptions while reading or writing to the file.
+    """
+    def exception_handle(*arguments):
+        try:
+            return function(*arguments)
+        except FileNotFoundError:
+            msg = f"File '{arguments[0]}' not found. Aborting"
+            logging.error(msg)
+            sys.exit(1)
+        except IOError as exception:
+            msg = f"I/O error '{exception.errno}': '{exception.strerror}'"
+            logging.error(msg)
+            sys.exit(1)
+        except Exception as err:
+            msg = f"Unexpected error while writing to the file: '{err}'"
+            logging.error(msg)
+            sys.exit(1)
+    return exception_handle
+
+
+@file_handle_error
+def read_input_file(input_file, db_name):
+    """
+    Input file function to read the file content and replace the database name.
+    """
+    file_data = {"dashboards": [], "datasources": []}
+    if PurePath(input_file).suffix == ".json":
+        with open(input_file, encoding="utf_8") as file_obj:
+            db_template = json.load(file_obj)
+            dashboards = db_template["dashboards"]
+            for dashboard in dashboards:
+                slices = dashboard["__Dashboard__"]["slices"]
+                for dashboard_slice in slices:
+                    db_params = json.loads(dashboard_slice["__Slice__"]["params"])
+                    # Replacing the database name with passed argument
+                    db_params["database_name"] = db_name
+                    dashboard_slice["__Slice__"]["params"] = json.dumps(db_params)
+
+                temp_dict = {"__Dashboard__": dashboard["__Dashboard__"]}
+                file_data["dashboards"].append(temp_dict)
+
+            datasources = db_template["datasources"]
+            for datasource in datasources:
+                db_params = json.loads(datasource["__SqlaTable__"]["params"])
+                # Replacing the database name with passed argument
+                db_params["database_name"] = db_name
+                datasource["__SqlaTable__"]["params"] = json.dumps(db_params)
+
+                temp_dict = {"__SqlaTable__": datasource["__SqlaTable__"]}
+                file_data["datasources"].append(temp_dict)
+    else:
+        raise argparse.ArgumentTypeError('File must have JSON extension')
+
+    return file_data
+
+
+@file_handle_error
+def write_output_file(output_file, file_data):
+    """
+    This function will write the file.
+    """
+    if PurePath(output_file).suffix == ".json":
+        with open(output_file, 'w', encoding="utf_8") as file_obj:
+            json.dump(file_data, file_obj, indent=4)
+    else:
+        raise argparse.ArgumentTypeError('File must have JSON extension')
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("database_name")
+    parser.add_argument("-i", "--input_file", type=str, default="pgd_monitoring_template.json")
```

Comment:
I would change this and the next line to set the type as a PurePath (or just Path), so you don't need to cast them in other functions.

Reply:
fixed.